dbms/src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp

```diff
@@ -216,6 +216,15 @@ void JSONEachRowRowInputFormat::readNestedData(const String & name, MutableColum
 
 bool JSONEachRowRowInputFormat::readRow(MutableColumns & columns, RowReadExtension & ext)
 {
+    /// Set flag data_in_square_brackets if data starts with '['.
+    if (!in.eof() && in.position() == in.buffer().begin()) {
```

Comment:
That's slightly incorrect. Because buffer is constantly refilled with new data and cursor is moved from the end of buffer to begin.